paddle/fluid/platform/dynload/warpctc.h

```diff
@@ -31,10 +31,15 @@ extern void* warpctc_dso_handle;
  * (for each function) to dynamic load warpctc routine
  * via operator overloading.
  */
+#ifndef _WIN32
+#define DECLARE_TYPE(__name, ...) ->decltype(__name(__VA_ARGS__))
```

Comment:
windows don't have decltype? really?

Reply:
VS2015 treat it bad when it is a function call with template arguments, leave it empty or give an decltype(auto) and the compiler would deduce it anyway.